modules/core/src/main/java/org/apache/ignite/internal/util/HostAndPortRange.java

```diff
@@ -53,60 +53,89 @@ public static HostAndPortRange parse(String addrStr, int dfltPortFrom, int dfltP
 
         String host;
 
+        String portStr;
         int portFrom;
         int portTo;
 
         if (F.isEmpty(addrStr))
             throw createParseError(addrStr, errMsgPrefix, "Address is empty");
 
-        final int colIdx = addrStr.indexOf(':');
+        if (addrStr.contains("[")) { // IPv6
```

Comment:
`[` and `]` are only needed when there is a port, otherwise IPv6 address does not have those.

Reply:
Added the case where there is IPv6 without `[` and `]` and port(s)